crowbar_framework/lib/crowbar/lock.rb

```diff
@@ -16,33 +16,14 @@
 #
 
 module Crowbar
-  class Lock
-    attr_accessor :logger, :path, :name
-    attr_reader :path, :name, :file
+  module Lock
+    autoload :Base,
```

Comment:
It's simply loading the classes only on demand if it's really required and not generally directly all at once.